src/Data/Profunctor/Unsafe.hs

```diff
@@ -86,7 +92,11 @@ infixl 8 .#
 -- 'lmap' (f '.' g) ≡ 'lmap' g '.' 'lmap' f
 -- 'rmap' (f '.' g) ≡ 'rmap' f '.' 'rmap' g
 -- @
+#if __GLASGOW_HASKELL__ >= 806
+class (forall a . Functor (p a)) => Profunctor p where
```

Comment:
This is the only meaningful change here. The rest of the code is to just make the package compile.

Reply:
This change should also discuss why `forall a b. (Coercible a b, Coercible c d) => Coercible (p a c) (p b d)` is *not* added. Given the `(.#)` and `(#.)` which essentially ask for that...

Reply:
And to answer myself. `Star f` instances will be a problem.

Reply:
That isn't the only caveat. You'd need to fix that all the way down to `Functor` and then folks who build 'moral' functor instances that use manual Map constructors will (rightly) complain.

The "right" fix to Functor and the better version of (#.) and (.#) is a rather hard sell.

You'd need to add something like

    liftCoercion :: Coercion a b => Maybe (Coercion (f a) (f b))

to `Functor` and the corresponding helpers to `Profunctor`, `Bifunctor`, etc. which can then handle cases like `Compose` which are otherwise problematic.

With that you can write 

`fmapCoerce :: Coercible a b => f a -> f b`

which can be constant time if everything in the functor composition chain allows it, without ruling out those "moral" functors, which pop up when folks do things like pipes and want to have manual mapping layers and 'lift' layers in their monad transformer stack and quotient them out in the interpreter.